apps/glusterfs/lookup.go

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"github.com/boltdb/bolt"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+)
+
+// LookupHandler is a cache & intermediary object used to
+// fetch IDs and objects from the db
+type LookupHandler struct {
+	tx *bolt.Tx
+}
+
+func Lookup(tx *bolt.Tx) *LookupHandler {
+	return &LookupHandler{tx: tx}
+}
+
+// CompleteVolumes returns a list of volume ID strings for volumes
+// that are completely created.
+func (lh *LookupHandler) CompleteVolumes() ([]string, error) {
+	return VolumeList(lh.tx)
+}
+
+// CompleteBricks returns a list of brick ID strings for bricks
+// that are completely created.
+func (lh *LookupHandler) CompleteBricks() ([]string, error) {
+	return BrickList(lh.tx)
+}
+
+// CompleteClusterInfo updates the given ClusterInfoResponse object so
+// that it only contains references to complete volumes, etc.
+func (lh *LookupHandler) CompleteClusterInfo(ci *api.ClusterInfoResponse) error {
```

Comment:
This api is slightly asymmetric compared to the other ones.
The others just mimic the VolumeList and BrickList functions,
and return lists of objects, but this one does not mimic the clusterinfo
or so but instead potentially modifies the provided info structure.

Also the nomenclature is not quite consistent:

* CompleteVolumes returns the list of complete volumes
* CompleteBrick returns the list of complete bricks
* CompleteClusterInfo does not return the complete cluster info, but instead modifies the provided cluster info to only list the complete objects... 
 
Not entirely sure what to propose here, but it somehow feels a little odd.